drud/ddev
# Add the option to add flags to custom commands, fixes #2491

## The Problem/Issue/Bug:
More and more bundled custom commands are providing flags but there is not option to describe this flags in the custom command using the cobra definition which prints the flags for a command.

## How this PR Solves The Problem:
A new option _Flags_ is recognized now to describe the flags with a JSON comment in the custom command.

Example:

`## Flags: [{"Name":"test","Shorthand":"t","Usage":"Usage of test"}]`

The _Shorthand_ is optional and can be omitted.

## Manual Testing Instructions:
Create a new custom command and add a flags definition to the comment header see above or the detailed explanation in the docs.

## Automated Testing Overview:
The provided tests fully cover all cases including errors and panics.

## Related Issue Link(s):
Closes #2491 

## Release/Deployment notes:
Nothing special is needed.
